lib/global-styles.php

```diff
@@ -206,6 +206,58 @@ function gutenberg_experimental_global_styles_get_theme() {
 	);
 }
 
+/**
+ * Given a block json, it returns an array of block selectors.
+ *
+ * @param array $block_json The block.json config.
+ * @return array
+ */
+function gutenberg_experimental_global_styles_extract_selectors( $block_json ) {
+	$selector        = $block_json['__experimentalSelector'];
+	$block_name      = $block_json['name'];
+	$block_selectors = array();
+	foreach ( $selector as $key => $value ) {
+		$block_selectors[ $block_name . '/' . $key ] = $value;
+	}
+	return $block_selectors;
+}
+
+/**
+ * Pulls the CSS selectors to use from the block.json config.
+ *
+ * @return array
+ */
+function gutenberg_experimental_global_styles_get_selectors() {
+	$selectors = array( 'global' => ':root' );
+
+	// We need to provide a mechanism for blocks to opt-in into this
+	// that can be used by 3rd party blocks as well.
+	// This list is an interim approach to avoid the performance cost of
+	// having to detect support by reading all core block's block.json.
+	$blocks_supported = array(
+		'paragraph',
+		'heading',
+	);
+
+	// The block library dir may not exist if working from a fresh clone => bail out early.
+	$block_library_dir = dirname( __FILE__ ) . '/../build/block-library/blocks/';
```

Comment:
> Are you sure it’s a good idea to read block definitions separately from their registration. 

No, it isn't! We need to read this data from the block registration, but we don't have that ability now so I had to hack something to demo how this would work. The problem, as you stated above, is that, right now, only dynamic blocks are registered in the server, so we'd need to change that.

> Why not allow to pass it there as a property in the settings param?

I guess the gist of this is to be able to get the block data from the block registry, so this code can be agnostic as to how it is registered (through `register_block_type` or `register_block_type_from_metadata`).

So, the action item here seems to be that we register all core blocks in the server, is that correct?

Reply:
Yes, it seems blocks that need a custom selector should be registered on the server. When no selector is available I guess we can use a class selector e.g: for "core/gallery" we can use ".wp-block-gallery" as the selector.